oasislmf/model_preparation/summaries.py

```diff
@@ -413,3 +419,155 @@ def generate_summaryxref_files(model_run_fp, analysis_settings):
+    :type exposure_summary: dict
+
+    :param peril: descriptive name of peril
+    :type peril: str
+
+    :param peril_code: OED peril code
+    :type peril_code: str
+
+    :param status: status returned by lookup ('success', 'fail' or 'nomatch')
+    :type status: str
+
+    :param loc_num: location number column heading from exposure file
+    :type loc_num: str
+
+    :return: populated exposure_summary dictionary
+    :rtype: dict
+    """
+
+    # Separate TIVs by coverage type and acquire sum
+    for coverage_type in SUPPORTED_COVERAGE_TYPES:
+        tiv_sum = df.loc[
+            (df['peril_id'] == peril_code) &
+            (df['coverage_type_id'] == SUPPORTED_COVERAGE_TYPES[coverage_type]['id']),
+            'tiv'
+        ].sum()
+        tiv_sum = float(tiv_sum)
+        exposure_summary[peril][status]['tiv_by_coverage'][coverage_type] = tiv_sum
+        if coverage_type in exposure_summary[peril]['all']['tiv_by_coverage']:
+            exposure_summary[peril]['all']['tiv_by_coverage'][coverage_type] += tiv_sum
+        else:
+            exposure_summary[peril]['all']['tiv_by_coverage'][coverage_type] = tiv_sum
+        exposure_summary[peril][status]['tiv'] += tiv_sum
+        exposure_summary[peril]['all']['tiv'] += tiv_sum
+
+    # Find number of locations
+    loc_count = df.loc[df['peril_id'] == peril_code, loc_num].drop_duplicates().count()
+    loc_count = int(loc_count)
+    exposure_summary[peril][status]['number_of_locations'] = loc_count
+    exposure_summary[peril]['all']['number_of_locations'] += loc_count
+
+    return exposure_summary
+
+
+@oasis_log
+def write_exposure_summary(
+    target_dir,
+    gul_inputs_df,
+    exposure_fp,
+    keys_errors_fp,
+    exposure_profile,
+    loc_num
+):
+    """
+    Create exposure summary as dictionary of TIVs and number of locations
+    grouped by peril and validity respectively. Writes dictionary as json file
+    to disk.
+
+    :param target_dir: directory on disk to write exposure summary file
+    :type target_dir: str
+
+    :param gul_inputs_df: dataframe from gul_inputs.get_gul_input_items(..)
+    :type gul_inputs_df: pandas.DataFrame
+
+    :param exposure_fp: file path to input exposure file
+    :type exposure_fp: str
+
+    :param keys_errors_fp: file path to keys erors file
+    :type keys_errors_fp: str
+
+    :param exposure_profile: profile defining exposure file
+    :type exposure_profile: dict
+
+    :param loc_num: location number column heading from exposure file
+    :type loc_num: str
+    """
+
+    # Get keys in GUL input items dataframe and merge with key errors to obtain
+    # status information
+    try:
+        gul_inputs_errors_df, _ = get_gul_input_items(
+            exposure_fp, keys_errors_fp, exposure_profile=exposure_profile
+        )
+    except OasisException:   # Empty dataframe
+        gul_inputs_errors_df = pd.DataFrame(columns=gul_inputs_df.columns)
+    keys_errors_df = pd.read_csv(keys_errors_fp)
```

Comment:
There is a wrapper method called `get_dataframe` in `oasislmf/utils/data.py`, which is recommended for loading dataframes from file. I would recommend setting data types explicitly for every column, which you can do via passing a data types dict.

        dtypes = {
            'locid': 'str',
            'perilid': 'str',
            'coveragetypeid': 'uint32',
            'status': 'str',
            'message': 'str'
    }

So line 517 can be replaced by

    keys_errors_df = get_dataframe(src_fp=keys_errors_fp, col_dtypes=dtypes)